modules/dhcp_isc/isc_state_changes_observer.rb

```diff
@@ -3,27 +3,132 @@ module ISC
     class IscStateChangesObserver
       include ::Proxy::Log
 
-      attr_reader :service, :leases_file, :config_file
+      class Events
+        attr_accessor :m, :last_event
 
-      def initialize(config_file, leases_file, subnet_service)
+        STARTED = :started
+        STOPPED = :stopped
+        MODIFIED = :modified
+        RECREATED = :recreated
+        NONE = :none
+
+        def initialize
+          @m = Monitor.new
+          @last_event = NONE
+        end
+
+        def started
+          push(STARTED)
+        end
+
+        def stopped
+          push(STOPPED)
+        end
+
+        def modified
+          push(MODIFIED)
+        end
+
+        def recreated
+          push(RECREATED)
+        end
+
+        def push(an_event)
+          m.synchronize do
+            case an_event
+            when STARTED
+              return false unless @last_event == NONE
+              @last_event = STARTED
+            when STOPPED
+              @last_event = STOPPED
+            when MODIFIED
+              return false unless @last_event == NONE
+              @last_event = MODIFIED
+            when RECREATED
+              return false if @last_event != NONE && @last_event != MODIFIED
+              @last_event = RECREATED
+            else
+              return false
+            end
+          end
+          return true
+        end
+
+        def pop
+          @m.synchronize do
+            to_return = @last_event
+            @last_event = NONE
+            to_return
+          end
+        end
+      end
+
+      attr_reader :service, :leases_file, :config_file, :events, :event_loop_active, :worker
+
+      def initialize(config_file, leases_file, subnet_service, events = Events.new)
         @config_file = config_file
         @leases_file = leases_file
         @service = subnet_service
+        @events = events
+        @event_loop_active = true
+        @worker = Thread.new { event_loop }
       end
 
-      def monitor_started
+      def start
+        events.started
+        worker.run
```

Comment:
Should this call `do_start` rather than queuing a STARTED event and using the worker? Currently the module is starting up immediately and then parsing the file in the background, while it used to be synchronous during module startup.